.github/workflows/ci.yml

```diff
@@ -0,0 +1,127 @@
+name: ci
+on:
+  push:
+    branches:
+      - "**"
+
+defaults:
+  run:
+    # NOTE: A bit stricter than the default bash options used by GitHub Actions
+    # (bash --noprofile --norc -e -o pipefail {0})
+    shell: bash --noprofile --norc -euo pipefail {0}
+
+# NOTE: Set concurrency for the current workflow to 1
+concurrency: ci-${{ github.ref }}-${{ github.workflow }}
+
+jobs:
+  build-and-deploy:
+    timeout-minutes: 60
+    runs-on: ubuntu-22.04
+    permissions:
+      actions: read
+      contents: read
+      id-token: write
+    steps:
+      - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+
+      - uses: capralifecycle/actions-lib/check-runtime-dependencies@72f450a706d380212723f38753a82d8bb40ea5fc # v1
+
+      - uses: capralifecycle/actions-lib/parse-config@72f450a706d380212723f38753a82d8bb40ea5fc # v1
+        id: config
+        with:
+          config-file: ".ldp.json"
+
+      - uses: actions/setup-node@b39b52d1213e96004bfcb1c61a8a6fa8ab84f3e8 # v4.0.1
+        with:
+          node-version: 16.20.2
```

Comment:
We should probably bump this

Reply:
Agreed, but the project doesn't build on Node 18 atm, and I didn't want to muddy up the GHA migration with lots of version bumping

Reply:
It doesn't? It works locally with Node v20 at least

Reply:
Yeah, works locally here too (both 18 and 20); I just reverted to the version it used previously after the errors that occurred during the build step [here](https://github.com/capralifecycle/liflig-repo-metrics/actions/runs/7756658205/job/21154415384). Might be unrelated, though

Reply:
As previously noted, after having tried again with Node 20 which also seem to cause problems, we're handling the version bump separately from the migration.